lib/dot/definitions.def

```diff
@@ -3,7 +3,7 @@
     var $lvl = it.level;
     var $dataLvl = it.dataLevel;
     var $schema = it.schema[$keyword];
-    var $schemaPath = it.schemaPath + '.' + $keyword;
```

Comment:
I did try this, but it gets copied directly to the generate code rather than being interpreted first.
i.e. you get a line in `lib\custom.js` that looks like:
```javascript
var $schemaPath = it.schemaPath + it.utils.getProperty($keyword);
```

when what we wanted is:
```javascript
var $schemaPath = it.schemaPath.foo; // or it.schemaPath['x-foo'] as appropriate
```

I couldn't work out how to make it insert the result of `getProperty()` rather than the literal value. So I just used hardcoded brackets notation directly as done in `custom.jst`:

```
23:   var {{=$definition}} = RULES.custom['{{=$keyword}}'].definition;
```

Can you suggest how to make this work (I'm not very familiar with dotjs)?

As an aside, is there a reason to favour one over the other?  I did do a quick search for performance comparisons, and while there was nothing conclusive it seemed that for fixed strings, brackets and dots are very similar in terms of performance.  Using variables with brackets notation is obviously slower as it can't optimise the lookup, but we are using a hardcoded string so it should be ok.